### PR TITLE
Get rid of partof cycles

### LINKED ISSUE #313: PR:000000001 and protein (CHEBI:36080) has cyclic dependency
Hi there, PR:000000001 and protein (CHEBI:36080) has cyclic dependency suggest removing the subclass relation. 

Thanks
Gabby

### PR DESCRIPTION
**Pull request by matentzn**
Fixes #313. 

This PR ensures that we never release OBA again with subclass of cycles (as a consequence of relaxing owl:equivalentClasses). 

Depends on 

- [x] https://github.com/obophenotype/bio-attribute-ontology/pull/331

@rays22 can you take this PR over? It is basically done. Remaining tasks:

- [x] Help Arwa fix the above and merge into branch
- [x] Run a test release on the branch and convince yourself I did not break anything major (looking at oba.owl in Protege, maybe look at the diff - whatever you feel is necessary).

### CODE REVIEW

File: src/ontology/oba.Makefile [hunk 1]
```diff
@@ -81,12 +81,15 @@ $(MIRRORDIR)/lipidmaps.owl: $(TEMPLATEDIR)/lipidmaps.tsv
 $(ONT)-full.owl: $(SRC) $(OTHER_SRC)
 	echo "INFO: Running FULL release, which is customised for OBA."
 	$(ROBOT) merge --input $< \
-		merge -i components/reasoner_axioms.owl \
```
Comment (matentzn) [inline src/ontology/oba.Makefile:84]: This is removing the following axiom:

```
ReflexiveObjectProperty(<http://purl.obolibrary.org/obo/BFO_0000050>)
```

Due to the global properties of this statement, we are dropping it.

File: src/ontology/oba.Makefile [hunk 2]
```diff
@@ -81,12 +81,15 @@ $(MIRRORDIR)/lipidmaps.owl: $(TEMPLATEDIR)/lipidmaps.tsv
...
-		materialize -T basic_properties.txt \
```
Comment (matentzn) [inline src/ontology/oba.Makefile:85]: We do this a bit later now.

File: src/ontology/oba.Makefile [hunk 3]
```diff
@@ -81,12 +81,15 @@ $(MIRRORDIR)/lipidmaps.owl: $(TEMPLATEDIR)/lipidmaps.tsv
...
-		reason --reasoner ELK --equivalent-classes-allowed all --exclude-tautologies structural \
+		reason --reasoner ELK --equivalent-classes-allowed asserted-only --exclude-tautologies structural \
+		remove --term PR:000000001 \
+			   --term SO:0000252 \
+			   --term SO:0000234 \
```
Comment (matentzn) [inline src/ontology/oba.Makefile:87]: We remove a number of asserted equivalent classes forcefully. All three terms have equivalent classes that we are preserving.

File: src/ontology/oba.Makefile [hunk 4]
```diff
@@ -81,12 +81,15 @@ $(MIRRORDIR)/lipidmaps.owl: $(TEMPLATEDIR)/lipidmaps.tsv
...
+		reason --reasoner ELK --equivalent-classes-allowed none --exclude-tautologies structural \
```
Comment (matentzn) [inline src/ontology/oba.Makefile:88]: This is only for QC purposes.

**Review by rays22 — APPROVED** [review]
1. I have created two mock releases with or without the changes, respectively,  in this PR.
oba.owl **with** the changes contains a lot more axioms of the type 
```
- SubClassOf(<http://purl.obolibrary.org/obo/XX_12345678> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/BFO_0000050> <...>)
```
that are related to `part of` something.
2. I have also checked the `oba.owl` release products in Protege carefully looking for major differences or problems in the two versions, but other than the extra `part of`-type subclasses (which make sense) I could not find anything outstanding.

:tada:

**Comment by matentzn** [comment]
THANKS! Feel free to merge, and when all other PRs are processed, create a release! Thank you so much for checking!

**Pull request merged**